typings/heap/index.d.ts

```diff
@@ -0,0 +1,83 @@
+// Patched definitions from DefinitelyTyped
```

Comment:
upstream pr?

Reply:
How does this differ from the DefinitelyTyped defs that we are removing in this PR?

Reply:
pop/shift did not have `| undefined` in the return type. When there are no more elements to pop or shift the return type is undefined.

I'll open a PR for updating the types in upstream.

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59184